core/variant/variant_parser.cpp

```diff
@@ -54,6 +54,15 @@ char32_t VariantParser::Stream::get_char() {
 	return get_char();
 }
 
+void VariantParser::StreamFile::set_file(const Ref<FileAccess> p_file) {
+	if (f == p_file) {
+		return;
+	}
+	f = p_file;
+
+	buffer_read_end_position = f->eof_reached() ? f->get_length() : f->get_position();
+}
```

Comment:
Use the old cache. Maybe a method that clears the cache and uses the new position of f is also needed? I'm not sure if it was necessary to add it.

Maybe this can be modified to clear the buffer when it is set.